adafruit_wm8960.py

```diff
@@ -20,49 +24,1617 @@
+        self._left_mic_volume_set = self._right_mic_volume_set = True
+
+    ## Zero Cross
+
+    left_mic_zero_cross = WOBit(_REG_LEFT_INPUT_VOLUME, 6)
+    right_mic_zero_cross = WOBit(_REG_RIGHT_INPUT_VOLUME, 6)
+
+    @property
+    def mic_zero_cross(self) -> bool:
+        return self.left_mic_zero_cross and self.right_mic_zero_cross
+
+    @mic_zero_cross.setter
+    def mic_zero_cross(self, value: bool) -> None:
+        self.left_mic_zero_cross = self.right_mic_zero_cross = value
+
+    ## Mute
+
+    _left_mic_mute = WOBit(_REG_LEFT_INPUT_VOLUME, 7)
+    _right_mic_mute = WOBit(_REG_RIGHT_INPUT_VOLUME, 7)
+
+    @property
+    def left_mic_mute(self) -> bool:
+        return self._left_mic_mute
+
+    @left_mic_mute.setter
+    def left_mic_mute(self, value: bool) -> None:
+        self._left_mic_mute = value
+        self._left_mic_volume_set = True
+
+    @property
+    def right_mic_mute(self) -> bool:
+        return self._right_mic_mute
+
+    @right_mic_mute.setter
+    def right_mic_mute(self, value: bool) -> None:
+        self._right_mic_mute = value
+        self._right_mic_volume_set = True
+
+    @property
+    def mic_mute(self) -> bool:
+        return self._left_mic_mute and self._right_mic_mute
+
+    @mic_mute.setter
+    def mic_mute(self, value: bool) -> None:
+        self._left_mic_mute = self._right_mic_mute = value
+
+    # Boost Mixer
+
+    _left_input2_boost = WOBits(3, _REG_INPUT_BOOST_MIXER_1, 1)
+
+    @property
+    def left_input2_boost(self) -> float:
+        value = self._left_input2_boost
+        if value == 0:
+            return None
+        # fmt: off
+        return map_range(
+            value, 1, 7,
+            BOOST_GAIN_MIN, BOOST_GAIN_MAX
+        )
+        # fmt: on
+
+    @left_input2_boost.setter
+    def left_input2_boost(self, value: float) -> None:
+        # fmt: off
+        self._left_input2_boost = (
+            0
+            if value < BOOST_GAIN_MIN
+            else round(map_range(
+                value, BOOST_GAIN_MIN, BOOST_GAIN_MAX,
+                1, 7
+            ))
+        )
+        # fmt: on
+
+    _right_input2_boost = WOBits(3, _REG_INPUT_BOOST_MIXER_2, 1)
+
+    @property
+    def right_input2_boost(self) -> float:
+        value = self._right_input2_boost
+        if value == 0:
+            return None
+        # fmt: off
+        return map_range(
+            value, 1, 7,
+            BOOST_GAIN_MIN, BOOST_GAIN_MAX
+        )
+        # fmt: on
+
+    @right_input2_boost.setter
+    def right_input2_boost(self, value: float) -> None:
+        # fmt: off
+        self._right_input2_boost = (
+            0
+            if value < BOOST_GAIN_MIN
+            else round(map_range(
+                value, BOOST_GAIN_MIN, BOOST_GAIN_MAX,
+                1, 7
+            ))
+        )
+        # fmt: on
+
+    @property
+    def input2_boost(self) -> float:
+        value = max(self._left_input2_boost, self._right_input2_boost)
+        # fmt: off
```

Comment:
Why disable formatting?

Reply:
I found that the Black formatting tool would rearrange the map_range parameters in odd ways. I've taken out these tags for now and I'm just letting Black do it's thing, but I'd be lying if I said I was always happy with how it handles formatting in more extreme cases.